persistence/src/main/java/edu/unc/lib/dl/persist/services/acl/PatronAccessAssignmentService.java

```diff
@@ -0,0 +1,342 @@
+                    .addEventDetail(eventText)
+                    .create();
+        }
+    }
+
+    private String createRoleEventDetails(Collection<RoleAssignment> assignments) {
+        StringBuilder details = new StringBuilder("Patron roles for item set to:");
+        details.append(NEWLINE);
+        if (assignments == null || assignments.isEmpty()) {
+            details.append("No roles assigned");
+        } else {
+            String roleDetails = assignments.stream()
+                    .map(a -> a.getPrincipal() + ": " + a.getRole().getPropertyString())
+                    .collect(Collectors.joining(NEWLINE));
+            details.append(roleDetails);
+        }
+
+        return details.toString();
+    }
+
+    private void assertAssignmentsComplete(PatronAccessDetails details) {
+        if (details.getRoles() == null) {
+            return;
+        }
+        for (RoleAssignment assignment: details.getRoles()) {
+            if (assignment.getRole() == null) {
+                throw new IllegalArgumentException("Assignment must provide a role");
+            }
+            if (isBlank(assignment.getPrincipal())) {
+                throw new IllegalArgumentException("Assignment must provide a principal");
+            }
+        }
+    }
+
+    private void assertOnlyPatronRoles(PatronAccessDetails details) {
+        if (details.getRoles() != null && details.getRoles().stream().anyMatch(a -> a.getRole().isStaffRole())) {
+            throw new ServiceException("Only patron roles are applicable for this service");
+        }
+    }
+
+    private String createEmbargoEventDetails(String existingEmbargo, String newEmbargo) {
+        if (existingEmbargo == null) {
+            if (newEmbargo == null) {
+                // No change
+                return null;
+            } else {
+                // Assigned a new embargo
+                return "Set an embargo that will expire " + newEmbargo;
+            }
+        } else {
+            if (newEmbargo == null) {
+                // Cleared an existing embargo
+                return "Embargo expiration date changed from " + existingEmbargo + " to no embargo.";
+            } else if (newEmbargo.equals(existingEmbargo)) {
+                // No change
+                return null;
+            } else {
+                // Change existing embargo
+                return "Embargo expiration date changed from " + existingEmbargo + " to " + newEmbargo;
+            }
+        }
+    }
+
+    private void writePremisEvents(RepositoryObject repoObj, Resource rolesEvent, Resource embargoEvent) {
+        PremisLogger logger = repoObj.getPremisLog();
+
+        if (rolesEvent != null) {
+            if (embargoEvent != null) {
+                logger.writeEvents(rolesEvent, embargoEvent);
+            } else {
+                logger.writeEvents(rolesEvent);
+            }
+        } else {
+            logger.writeEvents(embargoEvent);
```

Comment:
Should this be written only if embargoEvent isn't null?

Reply:
This method only gets called if one or the other event variable is not null:
https://github.com/UNC-Libraries/Carolina-Digital-Repository/blob/8cc36d993bca3dbeef81d22bb5c2cfb4a55580ba/persistence/src/main/java/edu/unc/lib/dl/persist/services/acl/PatronAccessAssignmentService.java#L116

But there's a lot going on in this class, so I'm going to add a comment to explain why its not checking for this.